api/app/src/main/kotlin/packit/controllers/PacketController.kt

```diff
@@ -82,28 +83,21 @@ class PacketController(
             .body(response.first)
     }
 
-    @GetMapping("/{name}/{id}/zip")
-    @PreAuthorize("@authz.canReadPacket(#root, #id, #name)")
-    // TODO: Verify authorization for accessing specific files: whether user has access to all files, or artefacts only.
-    fun downloadZip(
+    @GetMapping("/{id}/zip")
+    // TODO: authorize packet
+    // TODO: Verify differential authorization for artefacts vs non-artefact files
+    fun streamZip(
         @PathVariable id: String,
-        @PathVariable name: String,
-        @RequestParam(required = true) hashes: List<String>,
-        @RequestParam(required = true) filenames: List<String>,
-    ): ResponseEntity<StreamingResponseBody>
-    {
-        val filesWithNames = hashes.mapIndexed { index, hash ->
-            val fileResource = packetService.getFileByHash(hash, false, hash).first
-            fileResource to filenames[index]
-        }
+        @RequestParam hashes: List<String>,
+        request: HttpServletRequest,
```

Comment:
```suggestion
```

This is a GET request with no body associated with it. There's not really any reason why you should want access to the request stream.